src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/DefaultDriverExceptionHandler.java

```diff
@@ -1,36 +1,516 @@
-    return DatabaseException.Code.CLOSED_CONNECTION.get(
-        errVars(schemaObject, map -> map.put(TemplateVars.ERROR_MESSAGE, exception.getMessage())));
+  public RuntimeException handle(InvalidKeyspaceException exception) {
+    return DatabaseException.Code.UNKNOWN_KEYSPACE.get(errVars(schemaObject, exception));
+  }
+
+  // ========================================================================
+  // AllNodesFailedException and subclasses
+  // ========================================================================
+
+  @Override
+  public RuntimeException handle(AllNodesFailedException exception) {
+    // Should always be created with errors from calling each node, re-process the most important
+    // error
+
+    var highestPriority = findHighestPriority(exception).orElseGet(() -> null);
+
+    return switch (highestPriority) {
+        // found a node specific error that is a driver based
+      case DriverException e -> maybeHandle(e);
+        // this is a non-driver based exception, so map to generic unexpected driver error
+      case RuntimeException re ->
+          DatabaseException.Code.UNEXPECTED_DRIVER_ERROR.get(errVars(schemaObject, re));
+        // could not work out what the node error was OR this was a subclass of the
+        // AllNodesFailedException
+        // this will be the null case, but also need a default label
+      case null -> DriverExceptionHandler.super.handle(exception);
+      default -> DriverExceptionHandler.super.handle(exception);
+    };
+  }
+
+  @Override
+  public RuntimeException handle(NoNodeAvailableException exception) {
+    // this is a special case of AllNodesFailedException where no nodes were available
+    return DatabaseException.Code.FAILED_TO_CONNECT_TO_DATABASE.get(
+        errVars(schemaObject, exception));
+  }
+
+  // ========================================================================
+  // QueryValidationException and subclasses
+  // - this is a subclass CoordinatorException but that is abstract
+  // ========================================================================
+
+  // Following exceptions fall back to the {@link #handleUnhandled(DriverException)}:
+  //
+  // - **AlreadyExistsException**
+  //    - This should be handled by the subclasses of this handler because they know the specific
+  // entity we are trying to create.
+  //    - See {@link CreateTableDriverExceptionHandler} for an example.
+  //
+  // - **InvalidConfigurationInQueryException**
+  //    - Happens if we get the DDL command wrong.
+  //    - Should not happen.
+
+  /**
+   * this is the InvalidRequestException in the Cassandra code base, there are 300+ usages in it.
+   * generally means while the syntax is correct, what you are asking it to do is not possible e.g.
+   * add to a list, but the column is not a list. the Data API should prevent this from happening,
+   * but if it does, it is a bug and maybe the user can fix it.
```

Comment:
nit: I think if it's a bug, user cannot fix it, but can possibly work around it (change something to avoid hitting the bug)?